From ef2e7c4f059d76f85f7929e0bfa10cd9d48a4350 Mon Sep 17 00:00:00 2001
From: yangwei1 <yangwei1@eswincomputing.com>
Date: Thu, 28 Nov 2024 10:41:03 +0800
Subject: [PATCH 298/654] WIN2030-16678:fix:add hsp clock for usb

Changelogs:
	if no this clock,the usb control will not work

Change-Id: Ic716176305b0d27b7408fc96dcd1f09a50b2540a
Signed-off-by: yangwei1 <yangwei1@eswincomputing.com>
---
 .../riscv/boot/dts/eswin/eswin-win2030-die0-soc.dtsi | 12 ++++++++----
 .../riscv/boot/dts/eswin/eswin-win2030-die1-soc.dtsi | 12 ++++++++----
 2 files changed, 16 insertions(+), 8 deletions(-)

diff --git a/arch/riscv/boot/dts/eswin/eswin-win2030-die0-soc.dtsi b/arch/riscv/boot/dts/eswin/eswin-win2030-die0-soc.dtsi
index dd1fedbdbabe..baf952a80c8d 100644
--- a/arch/riscv/boot/dts/eswin/eswin-win2030-die0-soc.dtsi
+++ b/arch/riscv/boot/dts/eswin/eswin-win2030-die0-soc.dtsi
@@ -1977,8 +1977,10 @@ d0_usbdrd3_0: usb0@50480000 {
 			compatible = "eswin,win2030-dwc3";
 			#address-cells = <2>;
 			#size-cells = <2>;
-			clocks =<&d0_clock WIN2030_GATE_HSP_USB0_SUSPEND_CLK>;
-			clock-names = "suspend";
+			clocks =<&d0_clock WIN2030_GATE_HSP_USB0_SUSPEND_CLK>,
+					<&d0_clock WIN2030_CLK_HSP_ACLK>,
+					<&d0_clock WIN2030_CLK_HSP_CFG_CLK>;
+			clock-names = "suspend","aclk", "cfg_clk";
 			eswin,hsp_sp_csr = <&d0_hsp_sp_csr 0x800 0x808 0x83c 0x840>;
 			resets = <&d0_reset HSPDMA_RST_CTRL SW_USB0_VAUX_RSTN>;
 			reset-names = "vaux";
@@ -2014,8 +2016,10 @@ d0_usbdrd3_1: usb1@50490000  {
 			compatible = "eswin,win2030-dwc3";
 			#address-cells = <2>;
 			#size-cells = <2>;
-			clocks =<&d0_clock WIN2030_GATE_HSP_USB1_SUSPEND_CLK>;
-			clock-names = "suspend";
+			clocks =<&d0_clock WIN2030_GATE_HSP_USB1_SUSPEND_CLK>,
+					<&d0_clock WIN2030_CLK_HSP_ACLK>,
+					<&d0_clock WIN2030_CLK_HSP_CFG_CLK>;
+			clock-names = "suspend","aclk", "cfg_clk";
 			eswin,hsp_sp_csr = <&d0_hsp_sp_csr 0x900 0x908 0x93c 0x940>;
 			resets = <&d0_reset HSPDMA_RST_CTRL SW_USB1_VAUX_RSTN>;
 			reset-names = "vaux";
diff --git a/arch/riscv/boot/dts/eswin/eswin-win2030-die1-soc.dtsi b/arch/riscv/boot/dts/eswin/eswin-win2030-die1-soc.dtsi
index c05df53e7bf5..f0f24cad97a4 100644
--- a/arch/riscv/boot/dts/eswin/eswin-win2030-die1-soc.dtsi
+++ b/arch/riscv/boot/dts/eswin/eswin-win2030-die1-soc.dtsi
@@ -2042,8 +2042,10 @@ d1_usbdrd3_0: usb0@70480000 {
 			compatible = "eswin,win2030-dwc3";
 			#address-cells = <2>;
 			#size-cells = <2>;
-			clocks =<&d1_clock WIN2030_GATE_HSP_USB0_SUSPEND_CLK>;
-			clock-names = "suspend";
+			clocks =<&d1_clock WIN2030_GATE_HSP_USB0_SUSPEND_CLK>,
+					<&d1_clock WIN2030_CLK_HSP_ACLK>,
+					<&d1_clock WIN2030_CLK_HSP_CFG_CLK>;
+			clock-names = "suspend","aclk", "cfg_clk";
 			eswin,hsp_sp_csr = <&d1_hsp_sp_csr 0x800 0x808 0x83c 0x840>;
 			resets = <&d1_reset HSPDMA_RST_CTRL SW_USB0_VAUX_RSTN>;
 			reset-names = "vaux";
@@ -2079,8 +2081,10 @@ d1_usbdrd3_1: usb1@70490000 {
 			compatible = "eswin,win2030-dwc3";
 			#address-cells = <2>;
 			#size-cells = <2>;
-			clocks =<&d1_clock WIN2030_GATE_HSP_USB1_SUSPEND_CLK>;
-			clock-names = "suspend";
+			clocks =<&d1_clock WIN2030_GATE_HSP_USB1_SUSPEND_CLK>,
+					<&d1_clock WIN2030_CLK_HSP_ACLK>,
+					<&d1_clock WIN2030_CLK_HSP_CFG_CLK>;
+			clock-names = "suspend","aclk", "cfg_clk";
 			eswin,hsp_sp_csr = <&d1_hsp_sp_csr 0x900 0x908 0x93c 0x940>;
 			resets = <&d1_reset HSPDMA_RST_CTRL SW_USB1_VAUX_RSTN>;
 			reset-names = "vaux";
-- 
2.51.0

